core/arch/arm/mm/tee_mm.c

```diff
@@ -190,13 +196,13 @@ tee_mm_entry_t *tee_mm_alloc(tee_mm_pool_t *pool, size_t size)
 
 			if (remaining < size)
 				/* out of memory */
-				return NULL;
+				goto out;
 		}
 	}
 
 	nn = tee_mm_add(entry);
```

Comment:
I just realized that this is a real problem in the pager case as the malloc code is paged and may cause a page fault while a mm pool is locked and causing a deadlock.

Since `tee_mm_add()` is a static function it's easy to change it to take a preallocated `tee_mm_entry_t` as argument.